### ISSUE TITLE
nolibc/memcpy + memmove

### ISSUE THREAD

**Issue by palainp** [dinosaure/gilbraltar#16]
I'm fine with the same memcpy+memmove as the ones in ocaml-freestanding/nolibc. My current testing is:
```
$ opam pin
gilbraltar.~dev                   git  git+https://github.com/palainp/gilbraltar#nolibc
gilbraltar-linux.~dev             git  git+https://github.com/palainp/gilbraltar#nolibc
gilbraltar-toolchain.~dev         git  git+https://github.com/palainp/gilbraltar#nolibc
mirage-console-gilbraltar.~dev    git  git+https://github.com/palainp/gilbraltar#nolibc
mirage-gilbraltar.~dev            git  git+https://github.com/palainp/gilbraltar#nolibc
$ dune clean && dune build kernel8.img
$ md5sum kernel8.img 
8a83b4fe753409c01e298cd0beb3a6e3  kernel8.img
$ sudo mount /dev/mmcblk0p1 /mnt/
$ cat /mnt/config.txt 
core_freq_min=500
#enable_uart=1
kernel_old=1
$ sudo cp kernel8.img /mnt/
$ sudo umount /mnt
$ sudo minicom -D /dev/ttyUSB0 -b 115200
 _____ _ _ _           _ _                                                      
|   __|_| | |_ ___ ___| | |_ ___ ___                                            
|  |  | | | . |  _| .'| |  _| .'|  _|                                           
|_____|_|_|___|_| |__,|_|_| |__,|_|                                             
EL:1                                                                            
Interrupts: up                                                                  
MMU: ON                                                                         
RPi4: Memory map: 752 MB addressable:                                           
RPi4:   reserved @ (0x0 - 0xffffffffffffffff)                                   
RPi4:       text @ (0x0 - 0x40e5f)                                              
RPi4:     rodata @ (0x40e60 - 0x4738f)                                          
RPi4:       data @ (0x47390 - 0x1012f4f)                                        
RPi4:       heap >= 0x1013000 < stack < 0x2f000000                              
Hello World from OCaml!                                                         
IRQ                                                                             

```

**Comment by palainp** [comment]
I forgot:
```
$ ocaml --version
The OCaml toplevel, version 4.12.1
$ opam list | grep -E '^mirage '
mirage                    3.10.6      The MirageOS library operating system
```
EDIT: It also works with ocaml 4.13.1 update+upgrade, the md5sum for kernel8.img is `69c5f87de40892d1ecb47c22ba93044c`.

**Comment by dinosaure** [comment]
Thanks for this issue, we need to introspect why it did not work before: #12 was merged **after** #10 and it adds a naïve MMU support. @TheLortex said that it's may be an illegal instruction (generated by the compiler) which was not available until #12. It still can be an illegal alignment because we update the memory layout with #12.

I'm in favor to re-integrate `memcpy`/`memmove` from `ocaml-freestanding/nolibc` but we need to clearly understand the intial problem. I will try to spend a time on that in few weeks.

**Comment by palainp** [comment]
Done in #23 and thanks !
